srtcore/packet.cpp

```diff
@@ -561,10 +561,8 @@ CPacket* CPacket::clone() const
 {
     CPacket* pkt = new CPacket;
     memcpy((pkt->m_nHeader), m_nHeader, HDR_SIZE);
-    pkt->m_pcData = new char[m_PacketVector[PV_DATA].size()];
+    pkt->allocate(m_PacketVector[PV_DATA].size());
```

Comment:
```suggestion
    pkt->allocate(this->getLength());
```
(and same in the line below)
Using `this` to make it a bit more clear what's going on here.

Reply:
Good suggestion. Maybe even with an assertion:
```suggestion
    pkt->allocate(this->getLength());
    SRT_ASSERT(this->getLength() == pkt.getLength());
```